js.go

```diff
@@ -843,8 +843,8 @@ func (js *js) ChanSubscribe(subj string, ch chan *Msg, opts ...SubOpt) (*Subscri
 }
 
 // PullSubscribe creates a pull subscriber.
-func (js *js) PullSubscribe(subj string, opts ...SubOpt) (*Subscription, error) {
-	return js.subscribe(subj, _EMPTY_, nil, nil, opts)
+func (js *js) PullSubscribe(subj, durable string, opts ...SubOpt) (*Subscription, error) {
+	return js.subscribe(subj, _EMPTY_, nil, nil, append(opts, Durable(durable)))
```

Comment:
should we prevent here that nats.Durable is passes so that it is not ambiguous?

Reply:
Ooh, yeah. Good idea.

Reply:
I updated `nats.Durable` to check if it has already been set. If so, it returns an error. 

```go
func Durable(name string) SubOpt {
	return subOptFn(func(opts *subOpts) error {
		if opts.cfg.Durable != "" {
			return fmt.Errorf("nats: option Durable set more than once")
		}
		if strings.Contains(name, ".") {
			return ErrInvalidDurableName
		}

		opts.cfg.Durable = name
		return nil
	})
}
```

I also added this new test code.

```go
sub, err = js.PullSubscribe("bar", "foo", nats.Durable("bar"))
if err == nil {
	t.Fatalf("Unexpected success")
}
if err != nil && err.Error() != `nats: option Durable set more than once` {
	t.Errorf("Expected consumer in pull mode error, got %v", err)
}
```